libs/infinity_emb/infinity_emb/fastapi_schemas/data_uri.py

```diff
@@ -0,0 +1,239 @@
+import mimetypes
+import re
+import sys
+import textwrap
+from base64 import b64decode as decode64
+from base64 import b64encode as encode64
+from dataclasses import dataclass
+from typing import Any, Dict, MutableMapping, Optional, Tuple, TypeVar, Union
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+from urllib.parse import quote, unquote
+
+T = TypeVar("T")
+
+MIMETYPE_REGEX = r"[\w]+\/[\w\-\+\.]+"
+MIMETYPE_REGEX_AUDIO_IMAGE = r"(audio|image)\/[\w\-\+\.]+"
+_MIMETYPE_RE = re.compile("^{}$".format(MIMETYPE_REGEX_AUDIO_IMAGE))
+
+CHARSET_REGEX = r"[\w\-\+\.]+"
+_CHARSET_RE = re.compile("^{}$".format(CHARSET_REGEX))
+
+DATA_URI_REGEX = (
```

Comment:
@stikkireddy Does this regex work?

Reply:
[libs/infinity_emb/infinity_emb/transformer/vision/utils.py](https://github.com/michaelfeil/infinity/pull/385/files/a6175fb056cd39b17402f622f9874214e912fd6e#diff-a47efc277753063b843bf0543182a00f1918f50424d5f9b28877956bef7b4f19)

i assume you will remove that unnecessary code then?